src/cf.data.pre

```diff
@@ -4979,10 +4979,32 @@ DOC_START
 		[http::]<pt	Peer response time in milliseconds. The timer starts
 				when the last request byte is sent to the next hop
 				and stops when the last response byte is received.
-		[http::]<tt	Total time in milliseconds. The timer
-				starts with the first connect request (or write I/O)
-				sent to the first selected peer. The timer stops
-				with the last I/O with the last peer.
+		[http::]<tt	Total peering time in milliseconds.
```

Comment:
It is difficult for me to figure out the actual impact of inconsistencies alleged in this review because the current PR code also calls stopPeerClock(true) and because there are two timers (one in ALE and one in HttpRequest). This needs further research and probably some code changes (if not a total rewrite).

Answering the following partially overlapping questions may help you make progress:

1. Why does PR Client::serverComplete() still call stopPeerClock()? I thought we wanted FwdState (and TunnelStateData) to handle this timer starting from this PR because these classes know about retries and such (while Client does not). Also, Client does not startPeerClock(); it looks odd that it stops it.
2. Do we need to call stopPeerClock(true) somewhere? If not, we should remove the method parameter. If the only reason PR code needs that parameter is to accommodate totalResponseTime(), then we should drop the parameter and adjust totalResponseTime() (and other HierarchyLogEntry code) to function correctly without it -- we should not burden so many external callers to make one internal caller happier.
3. Should this (or a prerequisite!) PR eliminate HttpRequest::hier and keep ALE::hier? What do we need HttpRequest::hier for?

Reply:
> Why does PR Client::serverComplete() still call stopPeerClock()?

On one hand it looks like we should not call this method multiple times from a Client (probably several times due to retries) and call it once from FwdState destructor (or a similar terminating method). On the other, calling stopPeerClock(true) helps overcome the side effect of totalResponseTime(), outlined elsewhere. Can this be the reason why it has not been removed yet? If so, if can remove stopPeerClock(true) once we fix totalResponseTime().

> Should this (or a prerequisite!) PR eliminate HttpRequest::hier and keep ALE::hier?

There are lots of HttpRequest::hier occurrences and very few AccessLogEntry::hier. Also FwdState::syncHierNote()
and TunnelStateData::syncHierNote() update both fields at once so that, e.g., Acl::ServerCertificateCheck::match() could use AccessLogEntry::hier before it is copied from HttpRequest::hier in prepareLogWithRequestDetails(). I am not sure that we should do this refactoring in this PR.

Reply:
> Can this be the reason why it has not been removed yet?

I do not know the answer, but I suspect it is "no". Please note that my questions were about PR code. Sorry if I did not make it clear.  The history of these bugs may provide useful hints and increase our confidence, but given their volume and interdependence, I suspect that we should decide based on what fixed (i.e. this PR) code should do to support `%<tt` correctly (as specified in the updated documentation).

> If so, if can remove stopPeerClock(true) once we fix totalResponseTime().

Yes, I hope so.

>  I am not sure that we should [eliminate HttpRequest::hier] in this PR.

I agree. **If** it helps with this PR, we can eliminate HttpRequest::hier in a prerequisite PR. Otherwise, we should consider that for a followup PR.

Reply:
>  some code changes

Done (a71a763)

Reply:
> > history of these bugs may provide useful hints
> 
> startPeerClock()/stopPeerClock(bool) API was introduced a [16b70e2](https://github.com/measurement-factory/squid/commit/16b70e2a2dce20e73216f6ae1059152354df0b5f) as a fix for missing '%<tt' for tunnel/ssl_bump cases. It did not explain, however, the need for the 'bool' argument.

That 2014 commit predates Jira and GitHub. [squid-dev](http://www.squid-cache.org/mail-archive/squid-dev/201408/0149.html) [review](http://www.squid-cache.org/mail-archive/squid-dev/201409/0002.html) does not reveal any additional hints. It is not possible to recover that additional information without heroic efforts. Question closed.